src/Editor/GUI/Editors/TriggerEditor.cpp

```diff
@@ -0,0 +1,278 @@
+
+            switch (selectedTab) {
+            case 0: {
+                std::array<char, 100> name;
+                float delay = repeat->GetDelay();
+                float cooldown = repeat->GetCooldown();
+                int charges = repeat->GetTriggerCharges();
+                int startActive = repeat->GetStartActive();
+
+
+                memcpy(name.data(), repeat->GetName().c_str(), std::min(repeat->GetName().size(), name.size()));
+                name[std::min(repeat->GetName().size(), name.size() - 1)] = '\0';
+
+
+                // NAME
+                if (ImGui::InputText("Name", name.data(), name.size()))
+                    repeat->SetName(name.data());
+
+                // DELAY
+                if (ImGui::InputFloat("Delay", &delay, 0.1f, 1.0f, 1)) {
+
+                    if (delay < 0.0f)
+                        delay = 0.0f;
+
+                    repeat->SetDelay(delay);
+                }
+                ImGui::SameLine();
+                ImGui::Text("[?]");
+                if (ImGui::IsItemHovered())
+                    ImGui::SetTooltip("Time in seconds to delay.\nHold button to increase speed.");
+
+                // COOLDOWN
+                if (ImGui::InputFloat("Cooldown", &cooldown, 0.1f, 1.0f, 1)) {
+
+                    if (cooldown < 0.0f)
+                        cooldown = 0.0f;
+
+                    repeat->SetCooldown(cooldown);
+                }
+
+                ImGui::SameLine();
+                ImGui::Text("[?]");
+                if (ImGui::IsItemHovered())
+                    ImGui::SetTooltip("Time in seconds to cooldown.\nHold button to increase speed.");
+
+                // CHARGES                
+                if (ImGui::InputInt("Charges", &charges)) {
+
+                    if (charges < 0)
+                        charges = 0;
+
+                    repeat->SetTriggerCharges(charges);
+                }
+                ImGui::SameLine();
+
+                ImGui::Text("[?]");
+                if (ImGui::IsItemHovered())
+                    ImGui::SetTooltip("How many times a trigger can execute before automatically going inactive.\n0 = INFINITE.\nHold button to increase speed.");
+
+                // START ACTIVE
+                ImGui::Text("Start active? ");
+
+                ImGui::SameLine();
+
+                if (ImGui::RadioButton("Yes", &startActive, 1)) {
+                    repeat->SetStartActive(true);
+                }
+
+                ImGui::SameLine();
+
+                if (ImGui::RadioButton("No", &startActive, 0)) {
+                    repeat->SetStartActive(false);
+                }
+
+                break;
+            }
+            case 1: {
+
+                // Hardcoded single event for demonstration purposes
+                if (ImGui::Button("New event", ImVec2(100, 25))) {
+                    EventStruct newStruct;
+                    repeat->GetEventVector()->push_back(newStruct);
```

Comment:
Will this continually push structs to the vector?

Reply:
Everytime the button is clicked a new event will be pushed back. 
Only once per click.